### PR TITLE
Enable gradient checkpointing

### PR DESCRIPTION
**Pull request by woct0rdho**
This saves VRAM in training. The most naive way (do checkpoint at each layer) saves a lot of VRAM. It's also possible to do checkpoint every a few layers for a trade-off between time and space.

The signature of `diffusers.models.modeling_utils.ModelMixin._set_gradient_checkpointing` in recent versions of diffusers is different from that in your code, see https://github.com/huggingface/diffusers/blob/9836f0e000cfd826a7a5099002253ed2becc13e0/src/diffusers/models/modeling_utils.py#L1710 . I think there is no need to overload `_set_gradient_checkpointing` in your published code.

`use_reentrant=False` is recommended in recent versions of torch (>= 2.4). There is also no need to define `create_custom_forward`.

### CODE REVIEW

**Comment by woct0rdho** [comment]
Sorry, I found that this only works with `diffusers >= 0.33`. In `diffusers 0.32.2`, `_set_gradient_checkpointing` is not defined in `ModelMixin`, and we actually need to define it.

But anyway, it looks like a private API so I think it's better not to manually define it.

@ChuxiJ Do you think we can update diffusers to 0.33.1 in `requirements.txt`?
